### PR TITLE
Fix: specify the type

### PR DESCRIPTION
**Pull request by zzjc1234**
This pull request specifies the variable type to improve the code quality.

### CODE REVIEW

File: canvas_app.py [hunk 1]
```diff
@@ -183,11 +184,14 @@ async def delete_course(course_id: int):
         return JSONResponse(status_code=404, content={"message": "Courses not found"})
     courses = conf.get_conf()["courses"]
     all_courses = []
-    for course in courses:
-        if course["course_id"] != course_id:
-            all_courses.append(course)
-    conf.set_key_value("courses", all_courses)
-    return JSONResponse(status_code=200, content={"message": "success"})
+    if not isinstance(courses,Iterable):
```
Comment (linsyking) [inline canvas_app.py]: What does `not instance` do?

File: canvas_app.py [hunk 2]
```diff
@@ -229,14 +236,17 @@ async def create_course(course: Course):
...
-    for c in ori_courses:
-        if c["course_id"] == course.id and c["type"] == course.type:
-            return JSONResponse(
-                status_code=400, content={"message": "Course already exists"}
-            )
-    ori_courses.append(course_info)
-    conf.set_key_value("courses", ori_courses)
-    return JSONResponse(status_code=200, content={"message": "success"})
+    if not isinstance(ori_courses,Iterable):
+        JSONResponse(status_code=404, content={"message": "Not iterable"})
```
Comment (linsyking) [inline canvas_app.py]: "Not iterable" is not a very user-friendly error message

File: canvas_app.py [hunk 3]
```diff
@@ -249,6 +259,8 @@ async def modify_course(index: int, course: Course):
     if "courses" not in conf.get_conf():
         return JSONResponse(status_code=404, content={"message": "Courses not found"})
     courses = conf.get_conf()["courses"]
+    if not isinstance(courses,List):
+        return JSONResponse(status_code=404, content={"message": "Wrong Type"})
```
Comment (linsyking) [inline canvas_app.py]: Try making error message clearer :)

File: canvas_mgr.py [hunk 4]
```diff
@@ -58,7 +58,7 @@ def dump_out(self):
             json.dump(obj, f, ensure_ascii=False, indent=4)
         return self.g_out[:-1]
 
-    def print_own(self, mystr):
+    def print_own(self, mystr): #change the value of self.g_out
```
Comment (linsyking) [inline canvas_mgr.py]: `# Change ...`

**Review by linsyking — CHANGES_REQUESTED** [review]
Thanks for your contribution! You may need to modify the code a little bit :)

File: canvas_app.py (hunk 1 above)
Comment (zzjc1234) [reply canvas_app.py]: `not Isinstance` is designed for type checck. If coursse is not a iterable variable (actually, I think the type of courses should be a list of dictionary), then  the code should raise an error message.

File: canvas_mgr.py (hunk 4 above)
Comment (zzjc1234) [reply canvas_mgr.py]: ?

Comment (zzjc1234) [reply canvas_mgr.py]: Actually I don't understand what does the comment mean.

Comment (linsyking) [reply canvas_mgr.py]: Your comment format is not consistent with my code

Comment (zzjc1234) [reply canvas_mgr.py]: Ok,I'll fix it

Comment (zzjc1234) [reply canvas_mgr.py]: Currently I'm thinking about adding more comments to the code and I found a cool [plugin](https://github.com/folke/todo-comments.nvim). Is this kind of comment format acceptable?

Comment (linsyking) [reply canvas_mgr.py]: I mean the first letter should be capitalized and there should be a space

Comment (linsyking) [reply canvas_mgr.py]: The function specification should be like this:
```
def ddd():
    """
    Specification
    """

```

**Comment by linsyking** [comment]
[This](https://google.github.io/styleguide/pyguide.html) might be a good guideline for comment style in python

**Review by linsyking — APPROVED** [review]
Other enhancement should be in a different branch, merging this

**Comment by linsyking** [comment]
@zzjc1234 You may create a new branch to fix cq/format problem. This branch is for json-output feature.

**Comment by zzjc1234** [comment]
> @zzjc1234 You may create a new branch to fix cq/format problem. This branch is for json-output feature.

OK, I will take care next time.

**Pull request merged**